mlflow/tracing/trace_manager.py

```diff
@@ -0,0 +1,182 @@
+            return
+
+        trace_id = span.trace_id
+        if trace_id not in self._traces:
+            # NB: the first span might not be a root span, so we can only
+            # set trace_id here. Other information will be propagated from
+            # the root span when it ends.
+            self._create_empty_trace(trace_id)
+
+        trace_data_dict = self._traces[trace_id].span_dict
+        trace_data_dict[span.span_id] = span
+
+    def _create_empty_trace(
+        self,
+        trace_id: str,
+        experiment_id: Optional[str] = None,
+        attributes: Optional[Dict[str, str]] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ):
+        trace_info = TraceInfo(
+            trace_id=trace_id,
+            experiment_id=experiment_id
+            or "EXPERIMENT",  # TODO: Fetch this from global state or create a new one
+            start_time=None,
+            end_time=None,
+            status=Status(StatusCode.UNSET),
+            attributes=attributes or {},
+            tags=tags or {},
+        )
+
+        if trace_id in self._traces:
+            _logger.warning("Trace with ID {trace_id} already exists.")
+            return
+
+        with self._lock:
+            if trace_id not in self._traces:
+                self._traces[trace_id] = _Trace(trace_info, {})
+
+    def get_trace_info(self, trace_id: str) -> Optional[TraceInfo]:
+        """
+        Get the trace info for the given trace_id.
+        """
+        trace = self._traces.get(trace_id)
+        return trace.trace_info if trace else None
+
+    def get_span_from_id(self, trace_id: str, span_id: str) -> Optional[MLflowSpanWrapper]:
+        """
+        Get a span object for the given trace_id and span_id.
+        """
+        with self._lock:
+            trace = self._traces.get(trace_id)
+
+        return trace.span_dict.get(span_id) if trace else None
+
+    # NB: Caching as this requires a linear search over all spans in the trace and
+    #   the return value should not change for the same trace_id.
+    @lru_cache(maxsize=128)
+    def get_root_span_id(self, trace_id) -> Optional[str]:
+        """
+        Get the root span ID for the given trace ID.
+        """
+        with self._lock:
+            trace = self._traces.get(trace_id)
+
+        if trace:
+            for span in trace.span_dict.values():
+                if span.parent_span_id is None:
+                    return span.context.span_id
+
+        return None
+
+    def pop_trace(self, trace_id) -> Optional[Trace]:
+        """
+        Pop the trace data for the given trace_id and return it as a ready-to-publish Trace object.
+        """
+        with self._lock:
+            trace: _Trace = self._traces.pop(trace_id, None)
+
+        if trace is None:
+            return None
+
+        span_list = [span.to_mlflow_span() for span in trace.span_dict.values()]
+        return Trace(trace.trace_info, TraceData(spans=span_list))
```

Comment:
```suggestion
        if trace is not None:
            span_list = [span.to_mlflow_span() for span in trace.span_dict.values()]
            return Trace(trace.trace_info, TraceData(spans=span_list))
```

Reply:
Could we add a method in _Trace named `convert_to_mlflow_trace` to replace this?

Reply:
> Could we add a method in _Trace named convert_to_mlflow_trace to replace this?

What will be passed to that method?